lib/editor/components/pattern/PatternStopCard.js

```diff
@@ -246,12 +254,41 @@ class PatternStopContents extends Component<Props, State> {
     updatePatternStops(activePattern, patternStops)
   }
 
+  _onPickupOrDropOffChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const selectedOptionValue: number = parseInt(evt.target.value)
+    const {activePattern, index, updatePatternStops} = this.props
+    const patternStops = [...activePattern.patternStops]
+    patternStops[index][evt.target.id] = selectedOptionValue
```

Comment:
Since the `patternStops` variable is a shallow clone of `activePattern.patternStops`, the original objects are still used. Therefore, this overwrites existing data. The patternStop at the particular index should also be cloned to avoid this.

Reply:
> The patternStop at the particular index should also be cloned to avoid this.

This part of the comment still hasn't been addressed.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/668/commits/59d429a741b40aa19232203bcba3a83b14658b55.